mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/impl/MongoTimestampTypeDescriptor.java

```diff
@@ -0,0 +1,19 @@
+package org.hibernate.ogm.datastore.mongodb.type.impl;
+
+import java.sql.Timestamp;
+import java.util.Date;
+
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.JdbcTimestampTypeDescriptor;
+
+public class MongoTimestampTypeDescriptor extends JdbcTimestampTypeDescriptor {
+
+	@Override
+	public <X> Date wrap(X value, WrapperOptions options) {
+		Date wraped = super.wrap( value, options );
+		if ( Date.class.isInstance( wraped ) && !Timestamp.class.isInstance( wraped ) ) {
+			return new Timestamp( wraped.getTime() );
+		}
+		return wraped;
```

Comment:
I don't like this approach,  I think a better one is the one we used in  [`AbstractGeoJsonObjectGridTypeDescriptor`](https://github.com/hibernate/hibernate-ogm/blob/master/mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/type/impl/AbstractGeoJsonObjectGridTypeDescriptor.java).


This class should extend `GridTypeDescriptor` and override the `getExtractor` in a similar way. Once you have done this you can change`MongoTimestampGridType`  to something like:

	public MongoTimestampGridType() {
		super( MongoTimestampTypeDescriptor.INSTANCE, JdbcTimestampTypeDescriptor.INSTANCE );
	}, 

I would also rename `MongoTimestampGridType` to `JdbcTimestampGridType` and `MongoTimestampTypeDescriptor`
to `JdbcTimestampTypeDescriptor` and move them to core. We will need this for the other dialects as well.